### PR TITLE
Fix Mac installation

### PR DESCRIPTION
**Pull request by DoomHammer**
### Summary
Changes to recipe:  **qt/6.x.x**

#### Motivation
Previous PR: https://github.com/conan-io/conan-center-index/pull/24642 did not fix this issue https://github.com/conan-io/conan-center-index/issues/20574

The path is still wrong and doesn't work with downstream projects.

#### Details

* Set the correct path to the `Info.plist.app.in` folder.
* Removed the previous copy of this file.
* Added extra simple check to test_package

---
- [X] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [X] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [X] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by franramirez688 — COMMENTED** [review]
Hi @DoomHammer 

Thanks for this PR 👏 

I just saw the PR where this line was introduced: https://github.com/conan-io/conan-center-index/pull/24642/files#diff-c548829da9d38ae15a8cb4447140bc50ac3f38250d1f348decd9569dc3298b24R842-R843

I was wondering if we could also remove the lines from the `package()` method:

```python
            copy(self, "Info.plist.app.in", src=os.path.join(self.package_folder, "lib", "cmake", "Qt6", "macos"),
                                            dst=os.path.join(self.package_folder, "res", "macos"))
```
I just tried it locally and it worked by applying the diff suggested by @uilianries in the PR (https://github.com/conan-io/conan-center-index/pull/24642) changelog.

**Comment by DoomHammer** [comment]
I'm not sure I understand, can you rephrase?

**Comment by uilianries** [comment]
@DoomHammer Could you please specify the exactly error that you are trying to fix, with a reproducible scenario? I see you comment [here](https://github.com/conan-io/conan-center-index/issues/20574#issuecomment-2255647141), but still not enough to have your scenario, we need more details like: your Conan profile, Conan version and full build log, including command line used.

The PR #24642 was exactly to fix the MACOSX BUNDLE with CMake, so I'm afraid your PR will bring back that same bug.

**Comment by franramirez688** [comment]
> I'm not sure I understand, can you rephrase?

@DoomHammer Yes, let me rephrase it. This [PR](https://github.com/conan-io/conan-center-index/pull/24642) from @uilianries introduced this piece of code:

```python
    def package(self):
        if self.settings.os == "Macos":
            # ...
            copy(self, "Info.plist.app.in", src=os.path.join(self.package_folder, "lib", "cmake", "Qt6", "macos"),
                                            dst=os.path.join(self.package_folder, "res", "macos"))
        # ...
        if self.settings.os == "Macos":
            filecontents += f'set(__qt_internal_cmake_apple_support_files_path 
                                              "${{CMAKE_CURRENT_LIST_DIR}}/../../../res/macos")\n'
```
Then, what I'm saying is to change it as you're suggesting here, but also remove the `copy()` line: 
```python
    def package(self):
        if self.settings.os == "Macos":
            # ...
            # copy(self, "Info.plist.app.in", src=os.path.join(self.package_folder, "lib", "cmake", "Qt6", "macos"),
            #                                dst=os.path.join(self.package_folder, "res", "macos"))
        # ...
        if self.settings.os == "Macos":
            filecontents += f'set(__qt_internal_cmake_apple_support_files_path 
                                              "${{CMAKE_CURRENT_LIST_DIR}}/../../../lib/cmake/Qt6/macos")\n'
```

> The PR https://github.com/conan-io/conan-center-index/pull/24642 was exactly to fix the MACOSX BUNDLE with CMake, so I'm afraid your PR will bring back that same bug.

@uilianries I tested it locally and that `res` folder is not there. I applied the changes above and it's working fine (modifying the test package for sure).

**Review by franramirez688 — COMMENTED** [review]
@DoomHammer 

I added a commit adding some extra checks to the `test_package` to test this special case and removing that useless `copy`. Waiting for @jcar87 feedback before approving.

**Review by jcar87 — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Review by franramirez688 — APPROVED** [review]
